pkgs/by-name/mo/moneydance/package.nix

```diff
@@ -0,0 +1,56 @@
+{ lib, stdenv, fetchurl, makeWrapper, dpkg, openjdk21, openjfx21, }:
+let
+  version = "2023.3_5064";
+  hash = "sha256-t1jBZJ2sL0YDx9915UWgpCzKnpNq1Qz5i7ygGXDI5dk=";
+in stdenv.mkDerivation rec {
+  pname = "moneydance";
+  inherit version;
+
+  src = fetchurl {
+    url = "https://infinitekind.com/stabledl/${version}/moneydance_linux_amd64.deb";
+    inherit hash;
+  };
+
+  jdk = openjdk21.override {
+    enableJavaFX = true;
+  };
+
+  nativeBuildInputs = [ dpkg makeWrapper ];
+  buildInputs = [ jdk openjfx21 ];
+
+  dontConfigure = true;
+  dontUnpack = true;
+  dontBuild = true;
+  dontFixup = true;
+
+  installPhase = let
+    wrapperArgs = lib.concatMapStringsSep " " (flag: "--add-flags ${lib.escapeShellArg flag}") [
+      "-client"
+      "--add-modules"
+      "javafx.swing,javafx.controls,javafx.graphics"
+      "-classpath"
+      "'${placeholder "out"}/libexec/*'" # single-quotes to protect '*' from the shell
```

Comment:
Is that necessary with `escapeShellArgs`?

Reply:
Yes. The `escapeShellArg` call protects the `*` from `makeWrapper`, but it ends up unquoted in the generated script.

Reply:
Can we apply it twice then?

Reply:
Yes, that would work, but IMHO this is starting to get a bit over-engineered? How do you like this new revision? It's quite explicit and contains (I believe) the minimum required shell cleverness.